Fix woocommerce admin eslint command & errors

chihsuan opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/32310.

Fixes the woocommerce-admin lint command & errors

- Copied woocommerce-admin's .eslintrc.js to js packages
- Corrected paths in tsconfig
- Ignored core's js packages from woocommerce-admin's eslint
- Fixed JS & SCSS lint errors
- Add lint cmd to nx `project.json`

### How to test the changes in this Pull Request:

- Run `pnpm nx build woocommerce`
- Should run `pnpm nx lint woocommerce-admin --skip-nx-cache` successfully
- Should run `pnpm nx lint:js:packages woocommerce-admin --skip-nx-cache` successfully


### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

Dev: Fix woocommerce-admin lint command & errors

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

chihsuan committed:
Fix paths in tsconfig

chihsuan committed:
Fix client lint errors

chihsuan committed:
Fix js package lint errors

chihsuan committed:
Ignore core js packages from woo admin's eslint

chihsuan committed:
Add .eslintrc.js for js packages

chihsuan committed:
Fix reminder-bar.scss lint error

chihsuan committed:
Fix tsconfig path

chihsuan committed:
Add lint cmd to project.json

rjchow reviewed (commented):
@chihsuan, re: having a separate lint commands per package, I think that might be the better way to go - I tried running `pnpm nx affected --target=lint` in #32357, after making an intentional lint error in @woocommerce/components and it ran the lint command in the WCA client repo, which I think goes counter to what we are trying to use `nx` for.

chihsuan committed:
Add lint command to js packages

Update lint js commands

Update customer-effort-score/package.json

chihsuan committed:
Update pnpm-lock.yaml

chihsuan commented:
>  having a separate lint commands per package, I think that might be the better way to go I tried running pnpm nx affected --target=lint

@rjchow  This is a good point. I've added lint commands for each package. Also removed the package's lint check from wca's lint script.

chihsuan committed:
Add .eslintrc config to each packages

Update packages .eslintrc.js

Update components & admin-e2e-tests lint command

louwie17 commented on plugins/woocommerce-admin/tsconfig.json:
```diff
@@ -17,32 +17,32 @@
 			"./typings",
 			"./node_modules/@types",
-			"packages/**/node_modules/@types"
+			"../../packages/js/**/node_modules/@types"
```
🎉

louwie17 commented on packages/js/admin-e2e-tests/.eslintrc.js:
```diff
@@ -0,0 +1,78 @@
+	env: {
+		'jest/globals': true,
+	},
+	extends: [ 'plugin:@woocommerce/eslint-plugin/recommended' ],
```
Given all our plugins extend from our own `@woocommerce/eslint-plugin`, and most of these rules are the same across all the other packages, I think we should just move most of these rules to our `@woocommerce/eslint-plugin` package, which will remove a lot of duplicate code in our repo.
We probably would want to update the `recommended.js` file with these, given that is what we are using across all our packages.

Also it might be worth revisiting the [@worpress/eslint-plugin](https://github.com/WordPress/gutenberg/blob/%40wordpress/eslint-plugin%4011.1.0/packages/eslint-plugin/configs/recommended.js) rules to see if we have any overlapping rules that aren't necessary anymore. 
Unfortunately eslint doesn't allow for an easy way to figure out if there are duplicate rules.

louwie17 replied:
The duplicate rules could be done at a later time, I think updating the `recommended.js` with most of these rules would be better to do before merging this PR.

chihsuan replied:
>  I think we should just move most of these rules to our @woocommerce/eslint-plugin package, which will remove a lot of duplicate code in our repo.

This is a good idea! 💯

chihsuan replied:
I've moved most of these rules to our `@woocommerce/eslint-plugin` package and updated eslintrc.js for all packages.

louwie17 reviewed (dismissed):
Overall looks good, and I appreciate you fixing the lint errors at the same time :)

I do think we should update our `eslint-plugin` package with these new rules instead of copying the relatively large `eslintrc.js` to each js package (see comment below).

chihsuan committed:
Update eslint-plugin recommended rules

chihsuan committed:
Update woocommerce-admin/.eslintrc.js

chihsuan committed:
Add @wordpress/eslint-plugin to root package.json

rjchow committed:
tsconfig modifications

- moved and updated .vscode/tasks.json up to root level, and add exclusion to .gitignore
- added --pretty to ts:check in wca package.json (because otherwise there's no syntax highlighting in the output, not sure why this is needed with nx and not without)
- added references to root tsconfig.base.json
- update development.md with instructions for ts checking

rjchow reviewed (dismissed):
LGTM! Very nice work!

running `pnpm nx affected --target=lint` does what it's supposed to do, but the output isn't what I hoped for. Oh well, it's not within the scope of this PR to fix tens of thousands of lint errors 🤣
